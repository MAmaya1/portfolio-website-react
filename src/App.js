import React, { useState, useEffect } from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import styled from 'styled-components';
import ReactGa from 'react-ga';

// Import Components

import NavBar from './components/NavBar';
import SlideOutMenu from './components/SlideOutMenu';
import HomePage from './components/HomePage';
import PortfolioPage from './components/PortfolioPage';
import AboutMePage from './components/AboutMePage';
import ContactMePage from './components/ContactMePage';

// Styles

const AppWrapper = styled.div`
  background: rgba(31, 41, 58, 0.9);
  min-height: 100vh;
`

function App() {

  const [menuOpen, toggleMenu] = useState(false);

  useEffect(() => {
    // Configure Google Analytics
    ReactGa.initialize(`${process.env.REACT_APP_TRACKING_ID}`);
    // Report Page View
    ReactGa.pageview(window.location.pathname + window.location.searchS)
  }, []);

  return (
    <Router>
      {menuOpen && 
          <SlideOutMenu 
            menuOpen={menuOpen}
            toggleMenu={toggleMenu}
        />}
      <AppWrapper>
        <NavBar
          menuOpen={menuOpen}
          toggleMenu={toggleMenu}
        />
        <Switch>
          <Route exact path='/' component={HomePage}/>
          <Route path='/portfolio' component={PortfolioPage}/>
          <Route path='/about-me' component={AboutMePage}/>
          <Route path='/contact-me' component={ContactMePage}/>
          <Route component={HomePage}/>
        </Switch>
      </AppWrapper>
    </Router>
  );
}

export default App;
