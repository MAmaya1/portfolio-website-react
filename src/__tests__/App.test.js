import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

// Import Component
import App from '../App';

afterEach(cleanup);

describe('App', () => {
  it('should render App without crashing', () => {
    render(<App />);  
  });

  it('should match Snapshot', () => {
    const wrapper = render(<App />);
    expect(wrapper).toMatchSnapshot();
  });
})