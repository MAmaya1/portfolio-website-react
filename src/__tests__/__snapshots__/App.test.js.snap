// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-fzoiQi bmtcGV"
      >
        <div
          class="sc-AxjAm jUQGcy"
        >
          <i
            class="fas fa-bars fa-2x"
          />
          <div
            class="sc-AxirZ ckVgyD"
          >
            <a
              aria-current="page"
              class="active"
              href="/"
            >
              HOME
            </a>
            <a
              href="/portfolio"
            >
              MY WORK
            </a>
            <a
              href="/about-me"
            >
              ABOUT ME
            </a>
            <a
              href="/contact-me"
            >
              CONTACT
            </a>
          </div>
        </div>
        <div
          class="sc-AxhUy xfcSr"
        >
          <h1
            class="sc-AxgMl rwrkg"
          >
            Mario 
            <span>
              Amaya
            </span>
          </h1>
          <h2
            class="sc-AxheI hSuNJY"
          >
            React Developer
          </h2>
          <div
            class="sc-Axmtr hOhrNQ"
          >
            <a
              data-testid="linkedIn"
              href="https://www.linkedin.com/in/marioamayaweb"
              rel="noopener noreferrer"
              target="_blank"
            >
              <i
                class="fab fa-linkedin fa-2x"
              />
            </a>
            <a
              data-testid="github"
              href="https://www.github.com/mamaya1"
              rel="noopener noreferrer"
              target="_blank"
            >
              <i
                class="fab fa-github fa-2x"
              />
            </a>
            <a
              data-testid="email"
              href="mailto:mario.amaya.web@gmail.com"
            >
              <i
                class="far fa-envelope fa-2x"
              />
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-fzoiQi bmtcGV"
    >
      <div
        class="sc-AxjAm jUQGcy"
      >
        <i
          class="fas fa-bars fa-2x"
        />
        <div
          class="sc-AxirZ ckVgyD"
        >
          <a
            aria-current="page"
            class="active"
            href="/"
          >
            HOME
          </a>
          <a
            href="/portfolio"
          >
            MY WORK
          </a>
          <a
            href="/about-me"
          >
            ABOUT ME
          </a>
          <a
            href="/contact-me"
          >
            CONTACT
          </a>
        </div>
      </div>
      <div
        class="sc-AxhUy xfcSr"
      >
        <h1
          class="sc-AxgMl rwrkg"
        >
          Mario 
          <span>
            Amaya
          </span>
        </h1>
        <h2
          class="sc-AxheI hSuNJY"
        >
          React Developer
        </h2>
        <div
          class="sc-Axmtr hOhrNQ"
        >
          <a
            data-testid="linkedIn"
            href="https://www.linkedin.com/in/marioamayaweb"
            rel="noopener noreferrer"
            target="_blank"
          >
            <i
              class="fab fa-linkedin fa-2x"
            />
          </a>
          <a
            data-testid="github"
            href="https://www.github.com/mamaya1"
            rel="noopener noreferrer"
            target="_blank"
          >
            <i
              class="fab fa-github fa-2x"
            />
          </a>
          <a
            data-testid="email"
            href="mailto:mario.amaya.web@gmail.com"
          >
            <i
              class="far fa-envelope fa-2x"
            />
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
